Foundation/NSCFArray.swift

```diff
@@ -12,7 +12,9 @@ import CoreFoundation
 
 internal final class _NSCFArray : NSMutableArray {
     deinit {
+#if !os(Android)
```

Comment:
I'm pretty sure this is semantically incorrect. Can you explain why not this?

Reply:
Ahhh, yes… this does rather stand out as being an area that should be investigated. Basically if this call is made the app will crash further down the line due presumably to freeing invalid block or freeing them when they shouldn’t be. If made a few passes at trying to get to the bottom of this but had no luck. Avoiding making this call solves the problem. I’m not even sure self is a CFArray instance at this point as even just asking for the count will crash the app and have no idea where the pointer is coming from.

Reply:
This is resolved - see below.